src/sysv/systemd/set-nis-domainname.in

```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/sh
+
+source /etc/sysconfig/network
```

Comment:
Maybe test if this file exists before sourcing it. In Gentoo there is no such file

Reply:
Is there any alternative source for this information in Gentoo?

Reply:
There is yp-tools but it does only install /etc/yp/nicknames
The /etc/sysconfig dir is there though. 

There is an systemd file:
cat /etc/systemd/system/domainname.service.d/00gentoo.conf
# NIS domain to be set by domainname.service
[Service]
Environment="NISDOMAINNAME=local"

I guess /etc/sysconfig/network will do but I think you should test that the file
exists before you try to source it

Reply:
@joakim-tjernlund a bit of context is that we are working on the eventual removal of yp-tools in Fedora as all the NIS servers and client-side libraries removed or about to be removed. This was a slow process in past, Fedora removed client-side NIS support few years ago and finally FreeIPA removed NIS server emulator couple months ago.

Most likely Gentoo and other distributions will follow up eventually and yp-tools will get deprecated/removed too.

I think it makes sense to try if `/etc/sysconfig/network` exists and then source it. If it is not there or NISDOMAIN is not defined there, it can definitely be provided to `sssd.service` via override too. In any case, doing a check for the existence of `/etc/sysconfig/network` wouldn't hurt.

Reply:
Fixed in the current version.